src/game.cpp

```diff
@@ -13113,33 +13127,159 @@ int game::slip_down_chance( bool show_messages )
+    tripoint target_pos = examp;
+    target_pos.z -= fall.height_to_floor_or_furniture();
+    std::string target_disp_name = m.disp_name( target_pos );
+
+    climbing_aid::condition_list conditions = climbing_aid::detect_conditions( you, examp );
+
+    climbing_aid::aid_list aids = climbing_aid::list( conditions );
+
+    // Debug:
+    {
+        std::string cond_desc = string_format( "Climbing aid conditions: %d", conditions.size() );
+        for( climbing_aid::condition &cond : conditions ) {
+            cond_desc += "\n" + cond.category_string() + ": " + cond.flag;
+        }
+        cond_desc += string_format( "\n\nClimbing aids available: %d", aids.size() );
+        for( const climbing_aid *aid : climbing_aid::list_all( conditions ) ) {
+            cond_desc += "\n#" + std::to_string( climb_affordance_menu_encode( aid->id ) )
+                         + " " + aid->id.str() + ": " + string_format( aid->down.menu_text, target_disp_name );
+        }
+        cond_desc += string_format( "\n\n%d-level drop; %d until furniture; %d until creature.",
+                                    fall.height, fall.height_until_furniture, fall.height_until_creature );
+        popup( cond_desc );
+    }
+
+    for( const climbing_aid *aid : aids ) {
+        // Enable climbing aid unless it would deploy furniture in occupied tiles.
+        bool enable_aid = true;
+        if( aid->down.deploys_furniture() &&
+            fall.height_until_furniture < std::min( fall.height, aid->down.max_height ) ) {
+            // Can't deploy because it would overwrite existing furniture.
+            enable_aid = false;
+        }
+
+        int hotkey = aid->down.menu_hotkey;
+        if( hotkey == 0 ) {
+            // Deployables require a hotkey def and we only show one non-deployable; use 'c' for it.
+            hotkey = 'c';
+        }
+
+        // TODO LOCALIZE CLIMBING
+        cmenu.addentry( climb_affordance_menu_encode( aid->id ), enable_aid, hotkey,
+                        string_format( enable_aid ? aid->down.menu_text : aid->down.menu_cant, target_disp_name ) );
+    }
+}
+
+bool game::climb_down_menu_pick( const tripoint &examp, int retval )
+{
+    climbing_aid_id aid_id = climbing_aid_default;
+
+    if( climb_affordance_menu_decode( retval, aid_id ) ) {
+        climb_down_using( examp, aid_id );
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void game::climb_down( const tripoint &examp )
+{
+    uilist cmenu;
+    cmenu.text = _( "How would you prefer to climb down?" );
+
+    climb_down_menu_gen( examp, cmenu );
+
+    // If there would only be one choice, skip the menu.
+    if( cmenu.entries.size() == 1 ) {
+        climb_down_menu_pick( examp, cmenu.entries[0].retval );
+    } else {
+        cmenu.query();
+        climb_down_menu_pick( examp, cmenu.ret );
+    }
+}
+
+void game::climb_down_using(
+    const tripoint &examp,
+    climbing_aid_id aid_id,
+    bool deploy_affordance )
```

Comment:
The arguments to this function don't seem special enough to need 4 lines to themselves
```suggestion
void game::climb_down_using( const tripoint &examp, climbing_aid_id aid_id, bool deploy_affordance )
```

Reply:
Resolved in forthcoming commit to #68265.